src/db/sysdb_subid.c

```diff
@@ -0,0 +1,158 @@
+/*
+    Copyright (C) 2021 Red Hat
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <talloc.h>
+
+#include "db/sysdb_private.h"
+#include "db/sysdb_subid.h"
+
+#define SUBID_SUBDIR "subid_ranges"
+
+
+errno_t sysdb_store_subid_range(struct sss_domain_info *domain,
+                                const char *name,
+                                int cache_timeout,
```

Comment:
expiration_time?

Reply:
> * Please, provide some description to the first commit message.

Do you think it makes sense to keep 2 commits? I was going to squash them after review.


> * Why conditional build? Why is build disabled by default?

This is MVP of experimental feature.
Corresponding code in shadow-utils is merged upstream, but there is no upstream release available yet. We brought functionality to Fedora via patches. But I doubt other distributions will do the same. I.e. out of Fedora/RHEL, this is at the moment for enthusiasts who are willing to play with a new feature and who is capable to rebuild shadow-utils from sources.
Moreover, I anticipate significant changes might be required once we have initial feedback.

So... at the very least it's conditional to allow distributions to avoid installing plugin that they can't use yet.

Once shadow-utils upstream release is done and widely used, Podman patches released (https://github.com/containers/storage/pull/882), etc (so it is more or less easy to use), we can change default and build code / install plugin in sssd-ipa, IMO.

>     * We don't use two blank lines between function definitions.

And this (monolithic text) makes me sad :) I didn't see such restrictions in https://sssd.io/contrib/coding-style.html


>     * Can the range be set also for trusted domain user? If yes, this implementation doesn't seem to support it.

No, FreeIPA doesn't support it in MVP.

Reply:
Fixed.